### PR TITLE
uar-1516 add validation to review trust individual trustee page

### PR DESCRIPTION
**Pull request by HatiDev**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-1516

### Change description
API validation to be applied to any data related to the 'Individuals involved in the trust? page 

### Work checklist

- [X] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java [hunk 1]
```diff
@@ -160,10 +160,14 @@ private void validateTrustDetails(OverseasEntitySubmissionDto overseasEntitySubm
 
             if (!overseasEntitySubmissionDto.isForUpdateOrRemove()) {
                 // Note that this validation is only done for registrations
-                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
+                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext, overseasEntitySubmissionDto.isForUpdateOrRemove());
                 historicalBeneficialOwnerValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
                 trustCorporateValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
             }
+
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:167]: Blank line can go as can the `else if (...)` part. just `else {` will achieve the same.

Comment (HatiDev) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:167]: edited

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java [hunk 2]
```diff
@@ -160,10 +160,14 @@ private void validateTrustDetails(OverseasEntitySubmissionDto overseasEntitySubm
 
             if (!overseasEntitySubmissionDto.isForUpdateOrRemove()) {
                 // Note that this validation is only done for registrations
-                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
+                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext, overseasEntitySubmissionDto.isForUpdateOrRemove());
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:163]: I would hard-code a value of `false` for the final parameter here. Makes things clearer (its value must be `false` as processing has entered the block)

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java [hunk 3]
```diff
@@ -160,10 +160,14 @@ private void validateTrustDetails(OverseasEntitySubmissionDto overseasEntitySubm
...
                 historicalBeneficialOwnerValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
                 trustCorporateValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
             }
+
+            else {
+                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext, overseasEntitySubmissionDto.isForUpdateOrRemove());
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:169]: And similarly, a value of `true` here.

File: src/test/resources/overseas_entity_v_3_2.json [hunk 4]
```diff
@@ -783,7 +783,8 @@
             "locality" : "string"
           },
           "date_became_interested_person" : "2022-06-10",
-          "ceased_date" : "2024-02-29"
+          "ceased_date" : "2024-02-29",
+          "is_individual_still_involved_in_trust" : true
```
Comment (mwestacott) [inline src/test/resources/overseas_entity_v_3_2.json:787]: Minor, but I'd make the value here `false`, just so the data-set makes sense (you wouldn't have a ceased date if individual was still involved with the trust).

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java (hunk 2 above)
Comment (HatiDev) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java]: done

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java (hunk 3 above)
Comment (HatiDev) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java]: done

File: src/test/resources/overseas_entity_v_3_2.json (hunk 4 above)
Comment (HatiDev) [reply src/test/resources/overseas_entity_v_3_2.json]: changed

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java [hunk 5]
```diff
@@ -160,10 +160,15 @@ private void validateTrustDetails(OverseasEntitySubmissionDto overseasEntitySubm
...
             if (!overseasEntitySubmissionDto.isForUpdateOrRemove()) {
                 // Note that this validation is only done for registrations
-                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
+                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext, false);
                 historicalBeneficialOwnerValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
                 trustCorporateValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext);
             }
+
+            else {
+                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext, true);
```
Comment (dlloyd1-ch) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:169]: You can call the `trustIndividualValidator.validate() `method outside of the 'if' 'else' statement just passing in the `isForUpdateOrRemove` boolean value.
i.e.
```suggestion
                trustIndividualValidator.validate(overseasEntitySubmissionDto.getTrusts(), errors, loggingContext,  overseasEntitySubmissionDto.isForUpdateOrRemove());
```

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java]: I've amended this now.

**Review by dlloyd1-ch — APPROVED** [review]

**Review by lduranteau — APPROVED** [review]

**Review by mattch1 — APPROVED** [review]

**Pull request merged**